### DISCUSSION TITLE
Using json_contains_any for filtering, I found that the retrieval performance dropped significantly.

### DISCUSSION THREAD

**Discussion by Richard-lrg** [milvus-io/milvus#28020] [Q&A and General discussion]
Using json_contains_any for filtering, it was found that the time taken to retrieve the response increased by an order of magnitude (1s -> 10s+)
Does anyone have any experience in performance optimization or any suggestions on how to use it correctly?

**Comment by xiaofan-luan** [comment]
This is as expected.
1. For 2.3.2, you can try to use array type rather than json, which might be much faster than json
2. For 2.4, we are working on index for json/array and other scalar datatypes and this might be much faster than current solution

**Reply by grimmerk** [reply]
> For 2.4, we are working on index for json/array and other scalar datatypes and this might be much faster than current solution

@xiaofan-luan  This could be an awesome/killer feature, and please make it!

**Reply by xiaofan-luan** [reply]
This is already on the way.
Milvus 2.4 will support inverted index on any data types except for json.
Milvus 2.5 will add json inverted index support
